src/types.rs

```diff
@@ -638,6 +643,9 @@ pub struct AuthorizedIdentifier<'a> {
     /// The source identifier, missing any wildcard context
     pub identifier: &'a Identifier,
     /// Whether the identifier should be interpreted as a wildcard
+    ///
+    /// This is only relevant for DNS identifiers and must be false for other
+    /// types of identifiers (e.g. IP addresses).
```

Comment:
This comes from a combination of [RFC 8555 7.1.4](https://www.rfc-editor.org/rfc/rfc8555#section-7.1.4) where it says (emphasis mine):

> This field MUST be present and true for authorizations created as a result of a newOrder request       **containing a DNS identifier** with a value that was a wildcard domain name.  For other authorizations, it MUST be absent.

and [a verified errata](https://www.rfc-editor.org/errata/eid6364) that further clarified it could be omitted, or explicitly set `false`.